test/test_folder.py

```diff
@@ -0,0 +1,137 @@
+    This tests for a user with right access, a user without and a none user.
+    """
+    # Create the parent chain
+    F1, F2, privateFolder, F4 = _createParentChain(admin)
+
+    # Get the parent chain for a user who has access rights
+    parents = Folder().parentsToRoot(F4, user=admin)
+    for idx in range(1, 4):
+        assert parents[idx]['object']['name'] == 'F%i' % idx
+
+    # Get the parent chain for a user who doesn't have access rights
+    with pytest.raises(AccessException):
+        parents = Folder().parentsToRoot(F4, user=user)
+        for idx in range(1, 4):
+            if idx == 3:
+                assert parents[idx] is None
+            else:
+                assert parents[idx]['object']['name'] == 'F%i' % idx
+
+    # Get the parent chain for a none user
+    with pytest.raises(AccessException):
+        parents = Folder().parentsToRoot(F4, user=None)
+        for idx in range(1, 4):
+            if idx == 3:
+                assert parents[idx] is None
+            else:
+                assert parents[idx]['object']['name'] == 'F%i' % idx
+
+
+def testGetResourceByPath(server, admin, user):
+    # Create the parent chain
+    F1, F2, privateFolder, F4 = _createParentChain(admin)
+    # Test access denied response for access 'hidden folder' for user with access rights,
+    # user without and none user
+    resp = server.request(path='/resource/lookup',
+                          method='GET', user=admin,
+                          params={
+                              'path': '/user/%s/%s/%s/%s/%s' % (
+                                  admin['login'],
+                                  F1['name'],
+                                  F2['name'],
+                                  privateFolder['name'],
+                                  F4['name'])
+                          })
+    assertStatusOk(resp)
+    assert resp.json['name'] == F4['name']
+    assert ObjectId(resp.json['_id']) == F4['_id']
+    resp = server.request(path='/resource/lookup',
+                          method='GET', user=user,
+                          params={
+                              'path': '/user/%s/%s/%s/%s/%s' % (
+                                  admin['login'],
+                                  F1['name'],
+                                  F2['name'],
+                                  privateFolder['name'],
+                                  F4['name'])
+                          })
+    assertStatus(resp, 400)
+    resp = server.request(path='/resource/lookup',
+                          method='GET', user=None,
+                          params={
+                              'path': '/user/%s/%s/%s/%s/%s' % (
+                                  admin['login'],
+                                  F1['name'],
+                                  F2['name'],
+                                  privateFolder['name'],
+                                  F4['name'])
+                          })
+    assertStatus(resp, 400)
```

Comment:
This should be three separate tests.  You might consider [parametrizing](https://docs.pytest.org/en/latest/example/parametrize.html) them to deduplicate the code.

Reply:
I didn't succeed to use `admin` or `user` as a parameters with `pytest.mark.parametrize`. Do you think it's an issue to have some duplicate code in testing?

Reply:
That's okay if it's too awkward in this case.  It's just a good trick to keep in mind for the future.